### PR TITLE
Use an event channel to communicate with the block sync extension

### PR DESCRIPTION
**Pull request by sgkim126**

### CODE REVIEW

**Comment by sgkim126** [comment]
@majecty The last two commits(14f7cf320db1d616d14a41351c66299a2b476252 and 22892b6d61f82c3e875002bcb77ef4a9574cf991) makes network extensions be used only in the extension thread.
But it makes the tendermint e2e tests fail. I think it's because there is a deadlock. Would you check it?

**Comment by sgkim126** [comment]
@majecty 
I'll upload another PR to make CodeChain use an event channel for Tendermint extension. It needs some more refactoring because the current tendermint design is not suited to do it.

File: sync/src/block/extension.rs [hunk 1]
```diff
@@ -690,8 +708,48 @@ impl Extension {
...
+pub struct BlockSyncSender(EventSender<Event>);
+
+impl From<EventSender<Event>> for BlockSyncSender {
+    fn from(sender: EventSender<Event>) -> Self {
+        BlockSyncSender(sender)
+    }
+}
+
+impl ChainNotify for BlockSyncSender {
+    fn new_headers(
```
Comment (sgkim126) [inline sync/src/block/extension.rs]: @Kais-DkM Should ChainNotify be synchronous?
Is it okay not to wait for the result of the event?

Comment (Kais-DkM) [reply sync/src/block/extension.rs]: `new_headers` and `new_blocks` creates all the lock they need.
It should be safe to call them asynchronously.

**Pull request merged**